src/main/resources/templates/db_scripts/common/createRecordsViewFunction.sql

```diff
@@ -0,0 +1,324 @@
+        JOIN statistical_code_type sct ON sct.id = sc.statisticalcodetypeid
+  )
+	SELECT jsonb_agg(DISTINCT jsonb_build_object('id', sc.statCodeJsonb ->> 'id') ||
+							  jsonb_build_object('code', sc.statCodeJsonb ->> 'code') ||
+							  jsonb_build_object('name', sc.statCodeJsonb ->> 'name') ||
+							  jsonb_build_object('statisticalCodeType', sc.statCodeTypeJsonb ->> 'name') ||
+							  jsonb_build_object('source', sc.statCodeTypeJsonb ->> 'source'))
+	FROM jsonb_array_elements( $1 ) AS e,
+	 	 stat_codes sc
+	WHERE sc.statCodeId = (e ->> 0)::uuid;
+$$ LANGUAGE sql strict;
+
+-- Drop additional indexes
+DROP INDEX IF EXISTS ${myuniversity}_${mymodule}.item_pmh_metadata_updateddate_idx;
+DROP INDEX IF EXISTS ${myuniversity}_${mymodule}.holdings_record_pmh_metadata_updateddate_idx;
+DROP INDEX IF EXISTS ${myuniversity}_${mymodule}.audit_instance_pmh_createddate_idx;
+DROP INDEX IF EXISTS ${myuniversity}_${mymodule}.audit_holdings_record_pmh_createddate_idx;
+DROP INDEX IF EXISTS ${myuniversity}_${mymodule}.audit_item_pmh_createddate_idx;
+DROP INDEX IF EXISTS ${myuniversity}_${mymodule}.instance_pmh_metadata_updateddate_idx;
+
+-- Creates additional indexes
+CREATE INDEX IF NOT EXISTS instance_pmh_metadata_updateddate_idx ON ${myuniversity}_${mymodule}.instance ((strToTimestamp(jsonb -> 'metadata' ->> 'updatedDate')));
+CREATE INDEX IF NOT EXISTS item_pmh_metadata_updateddate_idx ON ${myuniversity}_${mymodule}.item ((strToTimestamp(jsonb -> 'metadata' ->> 'updatedDate')));
+CREATE INDEX IF NOT EXISTS holdings_record_pmh_metadata_updateddate_idx ON ${myuniversity}_${mymodule}.holdings_record ((strToTimestamp(jsonb -> 'metadata' ->> 'updatedDate')));
+CREATE INDEX IF NOT EXISTS audit_instance_pmh_createddate_idx ON ${myuniversity}_${mymodule}.audit_instance ((strToTimestamp(jsonb ->> 'createdDate')));
+CREATE INDEX IF NOT EXISTS audit_holdings_record_pmh_createddate_idx ON ${myuniversity}_${mymodule}.audit_holdings_record ((strToTimestamp(jsonb -> 'record' ->> 'updatedDate')));
+CREATE INDEX IF NOT EXISTS audit_item_pmh_createddate_idx ON ${myuniversity}_${mymodule}.audit_item ((strToTimestamp(jsonb -> 'record' ->> 'updatedDate')));
+
+-- Creates function returned instances with records updates/deleted/added in specified period o time
+CREATE OR REPLACE function ${myuniversity}_${mymodule}.get_updated_instance_ids_view(startDate                          timestamptz,
+                                                                                     endDate                            timestamptz,
+                                                                                     deletedRecordSupport               bool DEFAULT TRUE,
+                                                                                     skipSuppressedFromDiscoveryRecords bool DEFAULT TRUE,
+                                                                                     allowTieredUpdateDate              bool DEFAULT TRUE)
```

Comment:
I think this name  'allowTieredUpdateDate 'is not intuitive enough. Can you name it to something that better describes its purpose?